nbextensions/bulkImportCell/tabs/configure.js

```diff
@@ -158,21 +86,58 @@ define([
         }
 
         function buildFilePathWidget(node) {
-            const paramBus = runtime
-                .bus()
-                .makeChannelBus({ description: 'Parent comm bus for input widget' });
             // This is the key in the model that maps to the list of params for the current app.
-            const paramKey = `params.${fileType}`;
+            const paramBus = buildMessageBus(fileType, 'Parent comm bus for filePath widget');
+            paramBus.on('parameter-changed', (message) => {
+                updateModelParameterValue(fileType, FILE_PATH_TYPE, message);
+            });
+            paramBus.on('sync-data-model', (message) => {
+                if (message.values) {
+                    model.setItem(['params', fileType, FILE_PATH_TYPE], message.values);
+                }
+            });
 
+            /* Here, we need to
+             * 1. Get the list of file path params.
+             * 2. Get the initial parameters (this will be an array that's serialized in the model right now)
+             * 3. Pass those along to the filepathwidget
+             */
+            // this is an array of parameter ids from the current spec.
             const widget = FilePathWidget.make({
                 bus: paramBus,
                 workspaceId: runtime.workspaceId(),
-                initialParams: model.getItem(paramKey),
+                paramIds: model.getItem(['app', 'fileParamIds', fileType]),
```

Comment:
Now instead of the FPW guessing what its parameter ids are, we give it an array. We also give it the set of initial params that live in the cell's data model.